Sources/Components/NativeAdvert/NativeAdvertGridView.swift

```diff
@@ -0,0 +1,142 @@
+
+        settingsButton.topAnchor.constraint(equalTo: container.topAnchor),
+        settingsButton.trailingAnchor.constraint(equalTo: container.trailingAnchor),
+
+        imageView.topAnchor.constraint(equalTo: container.topAnchor),
+        imageView.leadingAnchor.constraint(equalTo: container.leadingAnchor),
+        imageView.heightAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: 1 / imageAspectRatio),
+    ]
+
+    private lazy var compactConstraints: [NSLayoutConstraint] = [
+        imageView.widthAnchor.constraint(equalTo: container.widthAnchor),
+
+        detailsContainer.topAnchor.constraint(equalTo: imageView.bottomAnchor, constant: .mediumSpacing),
+        detailsContainer.bottomAnchor.constraint(equalTo: container.bottomAnchor, constant: -.mediumSpacing),
+        detailsContainer.leadingAnchor.constraint(equalTo: container.leadingAnchor),
+        detailsContainer.trailingAnchor.constraint(equalTo: container.trailingAnchor)
+    ]
+
+    private lazy var regularConstraints: [NSLayoutConstraint] = [
+        imageView.widthAnchor.constraint(equalTo: container.widthAnchor, multiplier: 0.5),
+        imageView.bottomAnchor.constraint(equalTo: container.bottomAnchor),
+
+        detailsContainer.topAnchor.constraint(equalTo: container.topAnchor, constant: .mediumLargeSpacing),
+        detailsContainer.bottomAnchor.constraint(equalTo: container.bottomAnchor, constant: -.mediumLargeSpacing),
+        detailsContainer.leadingAnchor.constraint(equalTo: imageView.trailingAnchor, constant: .mediumLargeSpacing),
+        detailsContainer.trailingAnchor.constraint(equalTo: container.trailingAnchor, constant: -.mediumLargeSpacing)
+    ]
+
+    // MARK: - Init
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) not implemented")
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        addSubview(container)
+
+        container.addSubview(imageView)
+        container.addSubview(detailsContainer)
+
+        container.addSubview(settingsButton)
+
+        NSLayoutConstraint.activate(sharedConstraints)
+
+        setConstraints()
+        setColors()
+    }
+
+    public func configure(with model: NativeAdvertViewModel) {
+        if let imageUrl = model.mainImageUrl {
+            imageDelegate?.nativeAdvertView(setImageWithURL: imageUrl, onImageView: imageView)
+        }
+
+        detailsContainer.configure(with: model, andImageDelegate: imageDelegate)
+    }
+
+    private func setConstraints() {
+        if traitCollection.horizontalSizeClass == .regular {
+            NSLayoutConstraint.deactivate(compactConstraints)
+            NSLayoutConstraint.activate(regularConstraints)
+        } else {
+            NSLayoutConstraint.deactivate(regularConstraints)
+            NSLayoutConstraint.activate(compactConstraints)
+        }
+    }
+
+    private func setColors() {
```

Comment:
Same comment for the `setFonts`, `setColors` methods, maybe `setupSizeClassFonts`, `setupSizeClassColors` is better since it doesn't setup all colors and fonts

Reply:
Yes, I agree. It's probably less verbose to do that. I'll fix it!